### PR TITLE
Wildcard: `<Card />` migration

### PR DESCRIPTION
**Pull request by umpox**
closes https://github.com/sourcegraph/sourcegraph/issues/27681

## Description

Automated migration of the `card card-*` styles to the `<Card />` WIldcard component. Powered by this codemod: https://github.com/sourcegraph/codemod/pull/73 (This PR is basically super rough as it just required tweaking an existing codemod to migrate. Migration is priority here, will cleanup codemod afterwards)

This PR:
- Fully migrates usage of Bootstrap `card-*` classes to the Wildcard `<Card />` component
- Removes global Card styles and classNames
- Adds `CardFooter` Wildcard component
- Some styling tweaks

## Notes
- The original Card migration involved tweaking the UI to match the Wildcard redesign Card styles. It means there was a chance of some small regressions for the padding of our cards after this migration. I have checked every `CardBody` and `Card` that could be affected and removed some redundant padding styles!

### CODE REVIEW

File: client/web/src/components/WebStory.tsx [hunk 1]
```diff
@@ -38,14 +38,16 @@ export const WebStory: React.FunctionComponent<WebStoryProps> = ({
...
         <MockedStoryProvider mocks={mocks} useStrictMocking={useStrictMocking}>
-            <MemoryRouter {...memoryRouterProps}>
-                <Tooltip />
-                <Children
-                    {...breadcrumbSetters}
-                    isLightTheme={isLightTheme}
-                    telemetryService={NOOP_TELEMETRY_SERVICE}
-                />
-            </MemoryRouter>
+            <WildcardThemeContext.Provider value={{ isBranded: true }}>
```
Comment (umpox) [inline client/web/src/components/WebStory.tsx:41]: Note: This change causes a lot of changes in Chromatic. Going to move to a different PR

File: client/wildcard/src/components/Card/Card.tsx [hunk 2]
```diff
@@ -20,17 +17,12 @@ interface CardProps extends React.HTMLAttributes<HTMLDivElement> {
...
-    ...attributes
-}) => (
-    <Component
-        className={classNames(styles.card, className, variant === 'interactive' && styles.cardInteractive)}
-        {...attributes}
-    >
-        {children}
-    </Component>
-)
+export const Card = React.forwardRef(({ children, className, as: Component = 'div', ...attributes }, reference) => {
```
Comment (umpox) [inline client/wildcard/src/components/Card/Card.tsx:20]: Note: I have removed the `interactive` variant prop.

Reasoning: It seems better to me just to style the card when it is used as a `button` element, which is all we would use the interactive prop for, as far as I can tell.

Doing it this way also ensures nobody accidentally styles `div` card like it would be interactive

File: client/wildcard/src/components/Card/components/CardFooter.tsx [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+import styles from './CardFooter.module.scss'
+
+interface CardFooterProps {
+    /**
+     * Used to change the element that is rendered.
+     */
+    as?: React.ElementType
+}
+
+export const CardFooter = React.forwardRef(
```
Comment (umpox) [inline client/wildcard/src/components/Card/components/CardFooter.tsx:15]: Added `CardFooter`. We had usage in the application and this wasn't implemented in the original `Card` PR

**Review by tjkandala — APPROVED** [review]
